sdk-tests/src/test/java/io/dapr/it/state/GRPCStateClientIT.java

```diff
@@ -44,7 +48,45 @@ protected DaprClient buildDaprClient() {
     return daprClient;
   }
 
+  /** Tests where HTTP and GRPC behavior differ in Dapr runtime. **/
 
+  @Test
+  public void getStateStoreNotFound() {
+    final String stateKey = "key";
 
+    DaprClient daprClient = buildDaprClient();
 
+    // DaprException is guaranteed in the Dapr SDK but getCause() is null in HTTP while present in GRPC implementation.
+    assertThrowsDaprException(
+        "INVALID_ARGUMENT",
+        "INVALID_ARGUMENT: state store unknown state store is not found",
+        () -> daprClient.getState("unknown state store", new State(stateKey), byte[].class).block());
+  }
+
+  @Test
+  public void getStatesStoreNotFound() {
+    final String stateKey = "key";
+
+    DaprClient daprClient = buildDaprClient();
+
+    // DaprException is guaranteed in the Dapr SDK but getCause() is null in HTTP while present in GRPC implementation.
```

Comment:
This has to do with how HTTP handles exception vs gRPC. The contract is that it will always be DaprException. The cause attribute being present is not part of the contract.